src/server/system_services/system_store.js

```diff
@@ -658,125 +666,132 @@ class SystemStore extends EventEmitter {
      *
      */
     async make_changes(changes) {
-        const bulk_per_collection = {};
-        const now = new Date();
-        const last_update = now.getTime();
         let any_news = false;
-        dbg.log0('SystemStore.make_changes:', util.inspect(changes, {
-            depth: 5
-        }));
-
-        const get_collection = name => {
-            const col = COLLECTIONS_BY_NAME[name];
-            if (!col) {
-                throw new Error('SystemStore: make_changes bad collection name - ' + name);
-            }
-            return col;
-        };
-        const get_bulk = name => {
-            const bulk = bulk_per_collection[name] ||
-                mongo_client.instance().collection(name).initializeUnorderedBulkOp();
-            bulk_per_collection[name] = bulk;
-            return bulk;
-        };
+        let last_update;
+
+        await this._load_serial.surround(async () => {
```

Comment:
This means that we won’t have parallel make_changes and refreshes/loads will hold up make_changes.
This could mean a slow down in the performance of the system.
Just making sure that we truly need that surround, since loads/refreshes affect it.

Reply:
@jeniawhite That's right. 
Where do you think this has an impact? 
md_aggregator updates? 
Somewhere else?

Reply:
@jeniawhite Yes, this was the intention

Reply:
@guymguym I would assume multiple part uploads and mainly object update flows.
Did not deep dive and check it though.

Reply:
1. concurrent make_change might update the same documents which will lose information.
2. loading while writing is also problematic regarding the validity of the data that we try to read, especially when we do differential reads

Reply:
@jeniawhite we are talking about system store updates only

Reply:
@nb-ohad 
You are right regarding system_store updates. Object flow irrelevant.